shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java

```diff
@@ -60,11 +90,92 @@ public static String description() {
 
   @Override
   public int run(CommandLine cl) throws IOException {
-    JournalMasterClient jmClient = mMasterJournalMasterClient;
-    String serverAddress = cl.getOptionValue(ADDRESS_OPTION_NAME);
-    NetAddress address = QuorumCommand.stringToAddress(serverAddress);
+    NetworkAddressUtils.ServiceType serviceType =
+        NetworkAddressUtils.ServiceType.MASTER_RAFT;
+    final JournalMasterClient jmClient;
+    if (cl.hasOption(DOMAIN_OPTION_NAME)) {
+      String domainVal = cl.getOptionValue(DOMAIN_OPTION_NAME);
+      try {
+        JournalDomain domain = JournalDomain.valueOf(domainVal);
+        if (domain == JournalDomain.JOB_MASTER) {
+          jmClient = mJobMasterJournalMasterClient;
+          serviceType = NetworkAddressUtils.ServiceType.JOB_MASTER_RAFT;
+        } else {
+          jmClient = mMasterJournalMasterClient;
+        }
+      } catch (IllegalArgumentException e) {
+        throw new InvalidArgumentException(ExceptionMessage.INVALID_OPTION_VALUE
+            .getMessage(DOMAIN_OPTION_NAME,
+                Arrays.toString(JournalDomain.values())));
+      }
+    } else {
+      jmClient = mMasterJournalMasterClient;
+    }
+
+    mRaftJournalConf =
+        RaftJournalConfiguration.defaults(serviceType);
+    List<InetSocketAddress> addresses = mRaftJournalConf.getClusterAddresses();
+    Set<RaftPeer> peers = addresses.stream()
+        .map(addr -> RaftPeer.newBuilder()
+            .setId(RaftJournalUtils.getPeerId(addr))
+            .setAddress(addr)
+            .build()
+        )
+        .collect(Collectors.toSet());
+    mRaftGroup = RaftGroup.valueOf(RAFT_GROUP_ID, peers);
+
+    String strAddr = cl.getOptionValue(ADDRESS_OPTION_NAME);
+    NetAddress address = QuorumCommand.stringToAddress(strAddr);
+    InetSocketAddress serverAddress = InetSocketAddress
+        .createUnresolved(address.getHost(), address.getRpcPort());
+    List<RaftPeer> oldPeers = new ArrayList<>(mRaftGroup.getPeers());
+    // if you cannot find the address in the quorum, throw exception.
+    if (oldPeers.stream().map(RaftPeer::getAddress).noneMatch(addr -> addr.equals(strAddr))) {
```

Comment:
trim?

Reply:
You mean the `strAddr`? It is from option, so it cannot be surrounded by whitespace I think.